Automattic/pocket-casts-ios
# Implement ExPlat

| 📘 Part of: #1986
|:---:|

Fixes #1987 

This PR integrates ExPlat and runs an AA test to double check our implementation works as expected.

## To test

- CI must be 🟢 
- Check `p1722974042562459/1722961411.090999-slack` to set your bookmarklet
- Make sure your A8C proxy is set correctly and is running
- You must login with your A8C account
- Set a breakpoint into `ExPlatService` line `78` -> `let decoder = JSONDecoder()`
- Run the app
- When the app stops at the breakpoint
- Check if response request URL: it should be `https://public-api.wordpress.com/wpcom/v2/experiments/0.1.0/assignments/pocketcasts?_locale=en&experiment_names=pocketcasts_paywall_ios_aa_test&anon_id=YOUR_ID`
- Print the JSON result with `po String(data: (try? JSONSerialization.data(withJSONObject: (try? JSONSerialization.jsonObject(with: data))!, options: [.prettyPrinted, .sortedKeys]))!, encoding: .utf8)!`, checking if the body contains this:
```
"{\n  \"debug\" : {\n    \"backend_aa_result\" : \"request not sampled\"\n  },\n  \"ttl\" : 60,\n  \"variations\" : {\n    \"pocketcasts_paywall_ios_aa_test\" : null\n  }\n}"
```

## Checklist
